File: Bidapp Demo App - ObjectiveC/bidappPlayground/ViewController.m
//
//  ViewController.m
//  bidappPlayground
//
//  Created by Vasiliy Masnev on 30.01.2023.
//  Copyright 2023 bidapp. All rights reserved.
//

#import "ViewController.h"
#import "BannersTableViewController.h"
#import "FullscreenShowDelegate.h"
#import "BannerDelegate.h"
#import "FullscreenLoadDelegate.h"
#import "tools.h"
#import <bidapp/bidapp.h>

@implementation ViewController
{
    FullscreenLoadDelegate* loadDelegate;
    
    BIDInterstitial* interstitial;
    BIDRewarded* rewarded;
    
    BannerDelegate* bannerDelegate;
    BIDBannerView* bannerView;
    
    IBOutlet UIView* __weak bannerBaseView;
}

static NSMutableArray* delegates = nil;

-(void)viewDidLoad
{
    [super viewDidLoad];
    
    loadDelegate = [FullscreenLoadDelegate new];
    
    interstitial = [BIDInterstitial new];
    interstitial.loadDelegate = loadDelegate;
    
    rewarded = [BIDRewarded new];
    rewarded.loadDelegate = loadDelegate;
    
    bannerDelegate = [BannerDelegate new];
    
    bannerView = [BIDBannerView bannerWithFormat:BIDAdFormat.banner_320x50 delegate:bannerDelegate];
    [bannerBaseView addSubview:bannerView];
}

- (IBAction)onShowInterstitial:(id)sender
{
	if (!delegates)
	{
		delegates = [NSMutableArray new];
	}

	id interstitialDelegate = [[FullscreenShowDelegate alloc]initWithViewController:self];
	[delegates addObject:interstitialDelegate];
	
	[interstitial showWithDelegate:interstitialDelegate];
}

- (IBAction)onShowRewarded:(id)sender
{
	if (!delegates)
	{
		delegates = [NSMutableArray new];
	}

	id rewardedDelegate = [[FullscreenShowDelegate alloc]initWithViewController:self];
	[delegates addObject:rewardedDelegate];
	
	[rewarded showWithDelegate:rewardedDelegate];
}

- (IBAction)onShowBanners:(id)sender
{
	BannersTableViewController *bannersViewController = [[BannersTableViewController alloc] init];
	[self presentViewController:bannersViewController animated:YES completion:nil];
}

- (void)presentViewController:(UIViewController *)viewControllerToPresent animated:(BOOL)flag completion:(void (^)(void))completion
{
    [super presentViewController:viewControllerToPresent animated:flag completion:completion];
    
    [bannerView stopAutorefresh];
}

- (void)dismissViewControllerAnimated:(BOOL)flag completion:(void (^)(void))completion
{
    [super dismissViewControllerAnimated:flag completion:completion];
    
    [bannerView startAutorefresh:30.0];
}

@end

File: Bidapp Demo App - Swift/bidappPlayground/ViewController.swift
//
//  ViewController.swift
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

import UIKit
import bidapp

class ViewController : UIViewController {
    
    static var delegates:[FullscreenShowDelegate] = []
    
    var interstitial:BIDInterstitial? = nil
    var rewarded:BIDRewarded? = nil
    let loadDelegate = FullscreenLoadDelegate()
    
    var bannerView:BIDBannerView? = nil
    let bannerDelegate = BannerDelegate()
    @IBOutlet weak var bannerBaseView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        interstitial = BIDInterstitial()
        interstitial?.loadDelegate = loadDelegate
        
        rewarded = BIDRewarded()
        rewarded?.loadDelegate = loadDelegate
        
        bannerView = BIDBannerView.banner(with: BIDAdFormat.banner_320x50 as! BIDAdFormat, delegate: bannerDelegate)
        
        if let bannerView = bannerView {
            bannerBaseView.addSubview(bannerView)
        }
    }

    @IBAction func onShowInterstitial(_ sender: Any) {
        let interstitialDelegate = FullscreenShowDelegate(viewController: self)
        ViewController.delegates.append(interstitialDelegate)
        
        interstitial?.show(with: interstitialDelegate)
    }

    @IBAction func onShowRewarded(_ sender: Any) {
        let rewardedDelegate = FullscreenShowDelegate(viewController: self)
        ViewController.delegates.append(rewardedDelegate)
        
        rewarded?.show(with: rewardedDelegate)
    }

    @IBAction func onShowBanners(_ sender: Any) {
        let bannersViewController = BannersTableViewController()
        present(bannersViewController, animated: true)
    }
    
    override func present(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
        super.present(viewControllerToPresent, animated: flag, completion: completion)
        
        bannerView?.stopAutorefresh()
    }
    
    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
        super.dismiss(animated: flag, completion: completion)
        
        bannerView?.startAutorefresh(30.0)
    }
}
